test/test_energy_storage_device.py

```diff
@@ -451,34 +451,40 @@ def test_impedance(self):
                     omega.append(float(tmp[0]))
                     z_real.append(float(tmp[1]))
                     z_imag.append(float(tmp[2]))
-                else:
-                    # must be in between a section
-                    # if we have output, add it as a section
-                    if len(omega) != 0:
-                        omega_sections.append(omega)
-                        z_real_sections.append(z_real)
-                        z_imag_sections.append(z_imag)
-                        omega.clear()
-                        z_real.clear()
-                        z_imag.clear()
                 line = fout.readline()
 
-        # Assert the following relationships hold:
-        # (1) `Omega` decreases accross each section
-        # (2) `Z_real` increases accross each section
-        # (3) `z_imag` decreases accross each section
-        for sec in omega_sections:
-            # relationship (1)
-            for i in range(1, len(sec)-1):
-                self.assertTrue(sec[i] < sec[i-1])
-        for sec in z_real_sections:
-            # relationship (2)
-            for i in range(1, len(sec)-1):
-                self.assertTrue(sec[i] >= sec[i-1])
-        for sec in z_imag_sections:
-            # relationship (3)
-            for i in range(1, len(sec)-1):
-                self.assertTrue(sec[i] <= sec[i-1])
+        # extract the reference data 
+        ref_path = '/notebooks/test/reference_data/'
+        with open('%simpedance.out' % ref_path, 'r') as fin:
+            # read until the beginning of the data
+            line = fin.readline()
+            while line.find('(Rad/s)') == -1:
+                line = fin.readline()
+
+            # take only lines of output
+            while line != '':
+                if line.find('.') != -1:
+                    # first section not separated by commas
+                    if line.find(',') == -1:
+                        tmp = line.lstrip(' ').split()
+                    else:
+                        tmp = line.split(',')
+                    ref_omega.append(float(tmp[0]))
+                    ref_z_real.append(float(tmp[1]))
+                    ref_z_imag.append(float(tmp[2]))
+                line = fin.readline()
+
+        # Affirm that list sizes are equivalent
+        self.assertEqual(len(omega), len(ref_omega))
+        self.assertEqual(len(z_real), len(ref_z_real))
+        self.assertEqual(len(z_imag), len(ref_z_imag))
+
+        # If using same input parameters as in `reference_data`,
+        #     entries should be identical
+        for i in range(len(omega)):
+            self.assertEqual(omega[i], ref_omega[i])
```

Comment:
This may not work on another computer. The roundoff errors depends on the compiler and the architecture. So if you run the code on another computer or use a different compiler, the reference solution may be slightly different. You need to check that the value are almost equal. Ofc the tolerance can be very small.